.github/scripts/test_and_publish.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import subprocess, sys, os, tempfile
+from distantrs import Invocation
+from glob import glob
+
+INVOCATION_DATA_PATH = '/tmp/distant-rs-invocation.txt'
+
+# required to find listener.py
+this_path = os.path.abspath(os.path.dirname(__file__))
+test_script_path = f"{os.environ.get('GITHUB_WORKSPACE')}/renode/test.sh"
+inv_data_str = ""
+robot_name = glob("*.robot")[0].split(".")[0]
+
+with open(INVOCATION_DATA_PATH, 'r') as f:
+    inv_data_str = f.read()
+
+inv_data_lst = inv_data_str.split("--")
+
+i = Invocation(
+        invocation_id=inv_data_lst[0],
+        auth_token=inv_data_lst[1]
+        )
+
+i.announce_target(robot_name)
```

Comment:
I don't like this, but we need it to have artifacts...

Reply:
Yes, it's kinda lame. Normally we'd run a one big robot file once with tons of tests but in this case we're getting multiple output XMLs and stuffs like that so it is necessary.